### PR TITLE
Remove scheduler sampling profiler state

### PR DESCRIPTION
**Pull request by bvaughn**
No one has been using this data so there's no reason to collect it. Event log has been maintained and tests have been updated.

Follow up to #20832

### CODE REVIEW

**Review by gaearon — APPROVED** [review]
beautiful

File: packages/scheduler/src/SchedulerFeatureFlags.js [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 export const enableSchedulerDebugging = false;
 export const enableIsInputPending = false;
-export const enableProfiling = false;
+export const enableProfiling = __VARIANT__;
```
Comment (bvaughn) [inline packages/scheduler/src/SchedulerFeatureFlags.js:11]: Also this change should ensure we leave the feature off for OSS builds while still running tests to make sure it doesn't break FB builds.

Comment (Remyelder99) [reply packages/scheduler/src/SchedulerFeatureFlags.js:11]: Yes

Comment (Remyelder99) [reply packages/scheduler/src/SchedulerFeatureFlags.js:11]: Enable profiling

**Review by 7711188705 — COMMENTED** [review]
Screenshot_2024-01-10-07-14-48-662_com facebook katana

**Pull request merged**